dex/msgjson/types.go

```diff
@@ -297,6 +297,33 @@ func NewNotification(route string, payload interface{}) (*Message, error) {
 	}, nil
 }
 
+// Unmarshal unmarshals the Payload field into the provided interface.
+func (msg *Message) Unmarshal(payload interface{}) error {
+	return json.Unmarshal(msg.Payload, payload)
+}
+
+// UnmarshalResult is a convenience method for decoding the Result field of a
+// ResponsePayload.
+func (msg *Message) UnmarshalResult(result interface{}) error {
+	resp, err := msg.Response()
+	if err != nil {
+		return err
+	}
+	if resp.Error != nil {
+		return fmt.Errorf("rpc error: %d: %s", resp.Error.Code, resp.Error.Message)
+	}
+	return json.Unmarshal(resp.Result, result)
```

Comment:
It would be nice to distinguish the second error, which is not a failure of `UnmarshalResult`, from the other two.  It seems to be working now, but I suspect we'll have the need to handle the "rpc error" differently from the other two that are probably not recoverable.  No specific change requested here, just noting.

Reply:
I've been thinking that `msgjson.RPCError` should implement the `error` interface directly. Could then distinguish using `errors.As`. Caller can still call `(*Message).Response` if they need to tell the difference though. I'll keep thinking about it.